cfgrammar/src/lib/yacc/grammar.rs

```diff
@@ -287,60 +309,60 @@ impl YaccGrammar {
             prods:            prods.into_iter().map(|x| x.unwrap()).collect(),
             prod_precs:       prod_precs.into_iter().map(|x| x.unwrap()).collect(),
             implicit_nonterm: implicit_nonterm.and_then(|x| Some(nonterm_map[&x]))
-        }
+        })
     }
 
     /// Return the index of the end terminal.
-    pub fn eof_term_idx(&self) -> TIdx {
+    pub fn eof_term_idx(&self) -> TIdx<StorageT> {
         self.eof_term_idx
     }
 
     /// Return the productions for nonterminal `i`. Panics if `i` doesn't exist.
-    pub fn nonterm_to_prods(&self, i: NTIdx) -> &[PIdx] {
+    pub fn nonterm_to_prods(&self, i: NTIdx<StorageT>) -> &[PIdx<StorageT>] {
         &self.rules_prods[usize::from(i)]
     }
 
     /// Return the name of nonterminal `i`. Panics if `i` doesn't exist.
-    pub fn nonterm_name(&self, i: NTIdx) -> &str {
+    pub fn nonterm_name(&self, i: NTIdx<StorageT>) -> &str {
         &self.nonterm_names[usize::from(i)]
     }
 
     /// Return an iterator which produces (in no particular order) all this grammar's valid `NTIdx`s.
-    pub fn iter_nonterm_idxs(&self) -> Box<Iterator<Item=NTIdx>> {
-        Box::new((0..self.nonterms_len).map(NTIdx::from))
+    pub fn iter_nonterm_idxs(&self) -> Box<dyn Iterator<Item=NTIdx<StorageT>>> {
+        Box::new((0..self.nonterms_len).map(|x| NTIdx::from(x)))
```

Comment:
I think you can remove the closure here: `map(NTIDx::from)` or some such?

Reply:
I did try this after Clippy complained, but for reasons I don't fully understand this leads to an error:

```
   --> cfgrammar/src/lib/yacc/grammar.rs:332:9
    |
113 | impl<StorageT: PrimInt + Unsigned> YaccGrammar<StorageT> {
    |      --------- help: consider adding an explicit lifetime bound `StorageT: 'static`...
...
332 |         Box::new((0..self.nonterms_len).map(NTIdx::from))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
note: ...so that the type `std::iter::Map<std::ops::Range<u32>, fn(u32) -> idxnewtype::NTIdx<StorageT> {<idxnewtype::NTIdx<StorageT> as std::convert::From<u32>>::from}>` will meet its required lifetime bounds
   --> cfgrammar/src/lib/yacc/grammar.rs:332:9
    |
332 |         Box::new((0..self.nonterms_len).map(NTIdx::from))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error
```

To me the code looks equivalent before and after...

Reply:
Yes, I couldn't get this to work either :(

Reply:
I feel like this might be a type inference bug, but honestly I'm not sure.